notifications/src/main/java/org/openedx/notifications/presentation/settings/NotificationsSettingsFragment.kt

```diff
@@ -74,28 +96,70 @@ class NotificationsSettingsFragment : Fragment() {
                 val windowSize = rememberWindowSize()
 
                 val notificationsConfiguration by viewModel.notificationsConfiguration.collectAsState()
+                val showPermissionRequestDialog by viewModel.showPermissionRequestDialog.collectAsState()
+                val uiMessage by viewModel.uiMessage.collectAsState(null)
 
                 NotificationsSettingsScreen(
                     windowSize = windowSize,
                     notificationsConfiguration = notificationsConfiguration,
+                    showPermissionRequestDialog = showPermissionRequestDialog,
+                    uiMessage = uiMessage,
                     onBackClick = {
                         requireActivity().supportFragmentManager.popBackStack()
                     },
                     discussionPreferenceChanged = {
                         viewModel.setDiscussionNotificationPreference(it)
                     },
+                    onPositiveButtonClick = {
+                        viewModel.dismissPermissionDialog()
+                        requestNotificationPermissionOrNavigate()
+                    },
+                    onNegativeButtonClick = {
+                        viewModel.dismissPermissionDialog()
+                    }
                 )
             }
         }
     }
+
+    private fun requestNotificationPermissionOrNavigate() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+            if (ActivityCompat.shouldShowRequestPermissionRationale(
+                    requireActivity(),
+                    android.Manifest.permission.POST_NOTIFICATIONS
+                )
+            ) {
+                navigateToAppSettings()
+            } else {
+                pushNotificationPermissionLauncher.launch(
+                    android.Manifest.permission.POST_NOTIFICATIONS
+                )
+            }
+        } else {
+            navigateToAppSettings()
+        }
+    }
+
+    private fun navigateToAppSettings() {
+        val intent = Intent().apply {
+            action = Settings.ACTION_APPLICATION_DETAILS_SETTINGS
+            addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
+            data = Uri.parse("package:${requireContext().packageName}")
```

Comment:
IMO better to store this `package:${requireContext().packageName}` in some constant file.